src/geocat/comp/fourier_filters.py

```diff
@@ -0,0 +1,178 @@
+import math as m
+
+import numpy as np
+import xarray as xr
+
+
+def fourier_filter(signal,
+                   frequency,
+                   cutoff_frequency_low=0,
+                   cutoff_frequency_high=0,
+                   time_axis=0,
+                   low_pass=False,
+                   high_pass=False,
+                   band_pass=False,
+                   band_block=False):
+    """Filter a dataset by frequency. This function allowes for low_pass, high_
+    pass, band_pass, or band_block filtering of the data's freqency
+    representation.
+
+    Parameters
+    ----------
+    signal : numpy.ndarray, xr.DataArray,
+        n-dimensional dataset
+
+    frequency : float
+        sample frequency of dataset
+
+    cutoff_frequency_low : float, Optional
+        low frequency for cutting fourier transform, used by low_pass, band_pass, band_block
+
+    cutoff_frequency_high : float, Optional
+        high frequency for cutting fourier transform, used by low_pass, band_pass, band_block
+
+    time_axis : int, Optional
+        the time axis of the data set
+
+    low_pass : boolean, Optional
+        runs a low_pass filter on the data if set to True
+
+    high_pass : boolean, Optional
+        runs a high_pass filter on the data if set to True
+
+    band_pass : boolean, Optional
+        runs a band_pass filter on the data if set to True
+
+    band_block : boolean, Optional
+        runs a band_block filter on the data if set to True
+
+    Returns
+    -------
+    return_signal : numpy.ndarray, xr.DataArray
+        signal with specified filters applied
+
+    Examples
+    --------
+    Example 1: The tidal cycle needs to be removed from a 10/hr oceanic dataset,
+    (https://tidesandcurrents.noaa.gov/waterlevels.html?id=9415020&units=standard&bdate=20210101&edate=20210131&timezone=GMT&datum=MLLW&interval=6&action=data)
+
+    >>>import xarray as xr
```

Comment:
There should be a space here for all the docstring examples. Also, you're never importing the fourier function itself in the examples.
```suggestion
    >>> import xarray as xr
```

Reply:
done